### PR TITLE
fix nf-tests for gatk4spark

### PR DESCRIPTION
**Pull request by matthdsm**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/gatk4spark/applybqsr/tests/nextflow.config [hunk 1]
```diff
@@ -0,0 +1 @@
+docker.userEmulation = true
```
Comment (maxulysse) [inline modules/nf-core/gatk4spark/applybqsr/tests/nextflow.config:1]: this is getting deprecated, I don't think we should do that...

**Comment by maxulysse** [comment]
no baserecalibrator?

**Comment by matthdsm** [comment]
> no baserecalibrator?

Nope, that one's still on pytest with passing tests, don't want to break it

File: modules/nf-core/gatk4spark/applybqsr/tests/nextflow.config (hunk 1 above)
Comment (matthdsm) [reply modules/nf-core/gatk4spark/applybqsr/tests/nextflow.config:1]: I'm happy to fix it another way if you have one 😓

Comment (matthdsm) [reply modules/nf-core/gatk4spark/applybqsr/tests/nextflow.config:1]: @nf-core/maintainers does anyone know what this option does exactly? It seems to be the only that gets spark going

Comment (matthdsm) [reply modules/nf-core/gatk4spark/applybqsr/tests/nextflow.config:1]: It's been documented [here](https://github.com/nextflow-io/nextflow/discussions/4601), but there has to be more going on than that

Comment (matthdsm) [reply modules/nf-core/gatk4spark/applybqsr/tests/nextflow.config:1]: from git, `docker.userEmulation` add this to the command
```
'-u $(id -u) -e "HOME=${HOME}" -v /etc/passwd:/etc/passwd:ro -v /etc/shadow:/etc/shadow:ro -v /etc/group:/etc/group:ro -v $HOME:$HOME'
```

Comment (mahesh-panchal) [reply modules/nf-core/gatk4spark/applybqsr/tests/nextflow.config:1]: Exactly. Update `docker.runOptions` with what you found on Github. And add the group too.

Comment (mahesh-panchal) [reply modules/nf-core/gatk4spark/applybqsr/tests/nextflow.config:1]: Actually I'm not sure if some of that should be split between `runOptions` and `containerOptions`.

Comment (matthdsm) [reply modules/nf-core/gatk4spark/applybqsr/tests/nextflow.config:1]: Adding the whole string to `runOptions` fixed the issue, so I think I'll leave it at that for now

**Review by maxulysse — APPROVED** [review]

**Comment by matthdsm** [comment]
Thanks again Maxime!

**Pull request merged**